plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -363,79 +348,177 @@ fu_mediatek_scaler_device_get_firmware_version(FuDevice *device, GError **error)
 	if (!fu_memread_uint32_safe(st_res->data,
 				    st_res->len,
 				    2,
-				    &tmpversion,
+				    &version_raw,
 				    G_LITTLE_ENDIAN,
 				    error))
 		return FALSE;
-	fw_version = mediatek_scaler_device_version_to_string(tmpversion);
-	fu_device_set_version(device, fw_version);
+	fu_device_set_version_raw(FU_DEVICE(self), version_raw);
 	return TRUE;
 }
 
 static gboolean
-fu_mediatek_scaler_device_probe(FuDevice *device, GError **error)
+fu_mediatek_scaler_device_open(FuDevice *device, GError **error)
 {
 	FuMediatekScalerDevice *self = FU_MEDIATEK_SCALER_DEVICE(device);
-	g_autoptr(FuUdevDevice) udev_parent = NULL;
-	g_autoptr(FuDeviceLocker) locker = NULL;
-	g_autofree gchar *vendor_id = NULL;
-	g_autofree gchar *vid = NULL;
-	g_autofree gchar *pid = NULL;
-	g_autofree gchar *hw_ver = NULL;
 
-	/* FuUdevDevice->probe */
-	if (!FU_DEVICE_CLASS(fu_mediatek_scaler_device_parent_class)->probe(device, error))
+	/* proxy */
+	if (!fu_device_open(FU_DEVICE(self->i2c_dev), error))
```

Comment:
Ah, it seems the right way to hold the reference to a opened device rather having to open it again and again.  The remoteo session to the device somewhat dropped, I will test it out tomorrow when entered the lab.

Reply:
if you let the daemon do it using ->open() then it does the lifecycling for you. :)